src/test/java/org/jabref/logic/exporter/XmpPdfExporterTest.java

```diff
@@ -0,0 +1,176 @@
+        toral2006.setField(StandardField.TITLE, "A proposal to automatically build and maintain gazetteers for Named Entity Recognition by using Wikipedia");
+        toral2006.setField(StandardField.BOOKTITLE, "Proceedings of EACL");
+        toral2006.setField(StandardField.PAGES, "56--61");
+        toral2006.setField(StandardField.EPRINTTYPE, "asdf");
+        toral2006.setField(StandardField.OWNER, "Ich");
+        toral2006.setField(StandardField.URL, "www.url.de");
+
+        toral2006.setFiles(Arrays.asList(new LinkedFile("non-existing", "path/to/nowhere.pdf", "PDF")));
+
+        vapnik2000.setCitationKey("vapnik2000");
+        vapnik2000.setField(StandardField.TITLE, "The Nature of Statistical Learning Theory");
+        vapnik2000.setField(StandardField.PUBLISHER, "Springer Science + Business Media");
+        vapnik2000.setField(StandardField.AUTHOR, "Vladimir N. Vapnik");
+        vapnik2000.setField(StandardField.DOI, "10.1007/978-1-4757-3264-1");
+        vapnik2000.setField(StandardField.OWNER, "Ich");
+    }
+
+    /**
+     * Create a temporary PDF-file with a single empty page.
+     */
+    @BeforeEach
+    void setUp() throws IOException {
+        xmpPreferences = mock(XmpPreferences.class);
+        // The code assumes privacy filters to be off
+        when(xmpPreferences.shouldUseXmpPrivacyFilter()).thenReturn(false);
+        when(xmpPreferences.getKeywordSeparator()).thenReturn(',');
+
+        encoding = Charset.defaultCharset();
+
+        filePreferences = mock(FilePreferences.class);
+        when(filePreferences.getUser()).thenReturn(tempDir.toAbsolutePath().toString());
+        when(filePreferences.shouldStoreFilesRelativeToBib()).thenReturn(false);
+
+        importer = new PdfXmpImporter(xmpPreferences);
+        exporter = new XmpPdfExporter(xmpPreferences);
+
+        databaseContext = new BibDatabaseContext();
+        dataBase = databaseContext.getDatabase();
+
+        initBibEntries();
+        dataBase.insertEntry(olly2018);
+        dataBase.insertEntry(toral2006);
+        dataBase.insertEntry(vapnik2000);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideBibEntriesWithValidPdfFileLinks")
+    void successfulExportToAllFilesOfEntry(BibEntry bibEntryWithValidPdfFileLink) throws Exception {
+        assertTrue(exporter.exportToAllFilesOfEntry(databaseContext, encoding, filePreferences, bibEntryWithValidPdfFileLink, Arrays.asList(olly2018)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideBibEntriesWithInvalidPdfFileLinks")
+    void unsuccessfulExportToAllFilesOfEntry(BibEntry bibEntryWithValidPdfFileLink) throws Exception {
+        assertFalse(exporter.exportToAllFilesOfEntry(databaseContext, encoding, filePreferences, bibEntryWithValidPdfFileLink, Arrays.asList(olly2018)));
+    }
+
+    public static Stream<Arguments> provideBibEntriesWithValidPdfFileLinks() {
+        return Stream.of(Arguments.of(olly2018));
+    }
```

Comment:
For just a single test object, I don't think that you'll have to make a parameterized test.

Reply:
I agree, but it does not hurt having it parameterized (does it?) and this way it's easier to add tests in the future.

Reply:
Won't hurt. 👍